frontend/packages/topology/src/components/graph-view/Topology.tsx

```diff
@@ -230,12 +239,18 @@ const Topology: React.FC<TopologyProps &
   }, [model, visualization, visualizationReady]);
 
   React.useEffect(() => {
-    Promise.all(componentFactoriesPromises)
-      .then((res) => {
-        setComponentFactories(res);
-      })
-      .catch(() => {});
-  }, [componentFactoriesPromises]);
+    if (isFactoryComponentsResolved) {
+      Promise.all(componentFactoriesPromises)
+        .then((res) => {
+          setComponentFactories([
+            ...res,
+            ...dynamicComponentFactoryExtensions.map((f) => f.properties.getFactory),
+          ]);
+        })
+        .catch(() => {});
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [componentFactoriesPromises, isFactoryComponentsResolved]);
 
   React.useEffect(() => {
     if (componentFactoriesPromises.length && !componentFactories.length) {
```

Comment:
Since you've separated the static and dynamic extensions into different variables, this check is no longer valid as it only applies to one.

See other comment about using `useResolvedExtensions` for both static and dynamic plugins.